File: testClients/hannaClient/AI.cpp
#include "AI.h"
#include "util.h"

AI::AI(Connection* conn) : BaseAI(conn) {}

const char* AI::username()
{
  return "Shell AI";
}

const char* AI::password()
{
  return "password";
}

//This function is run once, before your first turn.
void AI::init(){}

Tile* AI::getTile(int x, int y)
{
  return & tiles[x*mapHeight() +y];
}

//This function is called each time it is your turn.
//Return true to end your turn, return false to ask the server for updated information.
bool AI::run()
{
  for(int i = 0; i < tiles.size(); i++) //spawn as many unit as possible
  {
    if (tiles[i].owner() == playerID() )
    {
      tiles[i].spawn(0);
    }
  }
  for (int i = 0; i < units.size(); i++)
  {
    if (units[i].owner() == playerID() )
    {
      units[i].move(units[i].x()+1, units[i].y() );
      if (units[i].y()+1 >= mapHeight())
      {
        units[i].dig(*getTile(units[i].x(), units[i].y()+1));
        units[i].fill(*getTile(units[i].x(), units[i].y()+1));
      }
    }
  }
  
  return true;
}

//This function is run once, after your last turn.
void AI::end(){}

File: testClients/LionClient/AI.py
#-*-python-*-
from BaseAI import BaseAI
from GameObject import *

class AI(BaseAI):
  """The class implementing gameplay logic."""
  @staticmethod
  def username():
    return "Shell AI"

  @staticmethod
  def password():
    return "password"

  ##This function is called once, before your first turn
  def init(self):
    pass

  ##This function is called once, after your last turn
  def end(self):
    pass

  ##This function is called each time it is your turn
  ##Return true to end your turn, return false to ask the server for updated information
  def run(self):
    myPumpStation = [tile for tile in self.tiles if tile.owner==self.playerID and tile.pumpID != -1]
    unitAt = dict()
    for unit in self.units:
        unitAt [(unit.x, unit.y)] = unit
    for tile in myPumpStation:
        if (tile.x, tile.y) not in unitAt:
            tile.spawn (2)
    return 1

  def __init__(self, conn):
    BaseAI.__init__(self, conn)
